src/main/java/frc/robot/commands/auto/AutoDrive.java

```diff
@@ -0,0 +1,98 @@
+package frc.robot.commands.auto;
+
+import com.pathplanner.lib.PathPlannerTrajectory;
+import com.pathplanner.lib.controllers.PPHolonomicDriveController;
+
+import edu.wpi.first.math.controller.HolonomicDriveController;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.DriverStation;
+import frc.robot.RobotState;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.drive.DriveSubsystem;
+
+public class AutoDrive extends CommandBase {
+
+    private DriveSubsystem drive;
+    private PathPlannerTrajectory path;
+
+    private PIDController xController, yController;
+    private PIDController thetaController;
+
+    //TODO: get pid values
+    private double kPX = 0, kIX = 0, kDX = 0, kPY = 0, kIY = 0, kDY = 0, kPTheta = 0, kITheta = 0, kDTheta = 0;
+
+    private PPHolonomicDriveController controller;
+    private DriverStation.Alliance alliance;
+
+    private Timer time = new Timer();
+
+    public AutoDrive(DriveSubsystem drive, PathPlannerTrajectory path) {
+
+        this.alliance = DriverStation.getAlliance();
+
+        this.drive = drive;
+        this.path = PathPlannerTrajectory.transformTrajectoryForAlliance(path, alliance);
+
+        xController = new PIDController(kPX, kIX, kPX);
+        yController = new PIDController(kPY, kIY, kDY);
+        thetaController = new PIDController(kPTheta, kITheta, kDTheta);
+        thetaController.enableContinuousInput(0, 2*Math.PI);
+
+        controller = new PPHolonomicDriveController(xController, yController, thetaController);
```

Comment:
These things should go in (and be managed by) the DriveSubsystem.

A Mode could be created in DriveSubsystem to differentiate path following mode vs human control vs any other modes we may want

Then path following commands like this one would simply input a path to the drive subsystem and it would follow it. Then in the end() method you would switch back to manual driving.